File: lc2849/cell_reachable.py
class Solution:
    def isReachableAtTime(self, sx: int, sy: int, fx: int, fy: int, t: int) -> bool:
        moves = [(-1, 1), (0, 1), (1, 1), (1, 0), (1, -1), (0, -1), (-1, -1), (-1, 0)]
        current = set()
        next = set()
        current.add((fx, fy))

        for _ in range(1, t + 1):
            for x, y in current:
                for x0, y0 in moves:
                    if x + x0 >= 1 and y + y0 >= 1:
                        next.add((x + x0, y + y0))

            current = next
            next = set()

        return (sx, sy) in current

File: lc2849/cell_reachable.cpp
#include <algorithm>
#include <cstdlib>
class Solution {
public:
  bool isReachableAtTime(int sx, int sy, int fx, int fy, int t) {
    int width = abs(sx - fx);
    int height = abs(sy - fy);

    int min_time = abs(sx - fx) + abs(sy - fy);

    if (sx == fx && sy == fy and t == 1) {
      return false;
    }

    return t >= std::max(width, height);
  }
};
